pynamodb/tests/integration/test_transaction_integration.py

```diff
@@ -0,0 +1,257 @@
+from datetime import datetime
+
+import config as cfg
+import pytest
+
+from pynamodb.exceptions import PutError, DoesNotExist
+
+from pynamodb.attributes import NumberAttribute, UnicodeAttribute, UTCDateTimeAttribute, BooleanAttribute
+from pynamodb.connection.transaction import TransactGet, TransactWrite
+
+from pynamodb.models import Model
+
+IDEMPOTENT_PARAMETER_MISMATCH = 'IdempotentParameterMismatchException'
+PROVISIONED_THROUGHPUT_EXCEEDED = 'ProvisionedThroughputExceededException'
+RESOURCE_NOT_FOUND = 'ResourceNotFoundException'
+TRANSACTION_CANCELLED = 'TransactionCanceledException'
+TRANSACTION_IN_PROGRESS = 'TransactionInProgressException'
+
+class User(Model):
+    class Meta:
+        region = 'us-east-1'
+        table_name = 'user'
+
+    user_id = NumberAttribute(hash_key=True)
+
+
+class BankStatement(Model):
+
+    class Meta:
+        region = 'us-east-1'
+        table_name = 'statement'
+
+    user_id = NumberAttribute(hash_key=True)
+    balance = NumberAttribute(default=0)
+    active = BooleanAttribute(default=True)
+
+
+class LineItem(Model):
+
+    class Meta:
+        region = 'us-east-1'
+        table_name = 'line-item'
+
+    user_id = NumberAttribute(hash_key=True)
+    created_at = UTCDateTimeAttribute(range_key=True, default=datetime.now())
+    amount = NumberAttribute()
+    currency = UnicodeAttribute()
+
+
+class DifferentRegion(Model):
+
+    class Meta:
+        region = 'us-east-2'
+        table_name = 'different-region'
+
+    entry_index = NumberAttribute(hash_key=True)
+
+
+TEST_MODELS = [
+    BankStatement,
+    DifferentRegion,
+    LineItem,
+    User,
+]
+
+
+@pytest.fixture(scope='class')
+def setup(request):
+    import os
+    ddb_url = os.getenv("PYNAMODB_INTEGRATION_TEST_DDB_URL", cfg.DYNAMODB_HOST)
+    # must ensure that the connection's host is the same as the models'
+    from pynamodb.connection.transaction import _CONNECTION
+    _CONNECTION.host = ddb_url
+
+    for m in TEST_MODELS:
+        m.Meta.host = ddb_url
+        if not m.exists():
+            m.create_table(
+                read_capacity_units=10,
+                write_capacity_units=10,
+                wait=True
+            )
+
+    def fin():
+        [_m.delete_table() for _m in TEST_MODELS if _m.exists()]
+
+    request.addfinalizer(finalizer=fin)
+
+
+def get_error_code(error):
+    return error.cause.response['Error'].get('Code')
+
+
+@pytest.mark.usefixtures('setup')
+@pytest.mark.ddblocal
+class TestTransaction:
```

Comment:
There's a lot of model setup going on per test, perhaps moving that to a common setup/breakdown would be helpful.

Reply:
the original idea was to use a `setup` fixture with `scope="class"` to create before/delete after. it works locally, but is failing in CI, so I'm gonna see if moving it to `setup_class`/`teardown_class` fixes it